client/payments/data/actions.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { apiFetch } from '@wordpress/data-controls';
+
+/**
+ * Internal dependencies
+ */
+import { ACTION_TYPES } from './action-types';
+import { API_NAMESPACE } from './constants';
+import { PaymentGateway, RestApiError, SelectorKeysWithActions } from './types';
+
+export function getPaymentGatewaysSuccess(
+	paymentGateways: PaymentGateway[]
+): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_SUCCESS;
+	paymentGateways: PaymentGateway[];
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAYS_SUCCESS,
+		paymentGateways,
+	};
+}
+
+export function setPaymentGateway(
+	paymentGateway: PaymentGateway
+): {
+	type: ACTION_TYPES.SET_PAYMENT_GATEWAY;
+	paymentGateway: PaymentGateway;
+} {
+	return {
+		type: ACTION_TYPES.SET_PAYMENT_GATEWAY,
+		paymentGateway,
+	};
+}
+
+export function setError(
+	selector: SelectorKeysWithActions,
+	error: RestApiError
+): {
+	type: ACTION_TYPES.SET_ERROR;
+	selector: SelectorKeysWithActions;
+	error: RestApiError;
+} {
+	return {
+		type: ACTION_TYPES.SET_ERROR,
+		selector,
+		error,
+	};
+}
+
+export function setIsRequesting(
+	selector: SelectorKeysWithActions,
+	isRequesting: boolean
+): {
+	type: ACTION_TYPES.SET_IS_REQUESTING;
+	selector: SelectorKeysWithActions;
+	isRequesting: boolean;
+} {
+	return {
+		type: ACTION_TYPES.SET_IS_REQUESTING,
+		selector,
+		isRequesting,
+	};
+}
+
+export function* updatePaymentGateway(
+	id: string,
+	data: Partial< PaymentGateway >
+) {
+	yield setIsRequesting( 'updatePaymentGateway', true );
+
+	try {
+		const response: PaymentGateway = yield apiFetch( {
+			method: 'PUT',
+			path: API_NAMESPACE + '/payment_gateways/' + id,
+			body: JSON.stringify( data ),
+		} );
+
+		if ( response && response.id === id ) {
+			// Update the already loaded payment gateway list with the new data
+			yield setPaymentGateway( response );
```

Comment:
What do you think about using event naming here instead: `UPDATE_PAYMENT_GATEWAY_SUCCESS`

Reply:
🤦 I used a setter naming again. Totally agree with UPDATE_PAYMENT_GATEWAY_SUCCESS. I will make an update.

Reply:
@joshuatf updated in https://github.com/woocommerce/woocommerce-admin/pull/6918/commits/53a61cb02a2b3d49fad572b5853db829c8f74860